### ISSUE TITLE
Gradually increase of the latency in Java Sound during executions

### ISSUE THREAD

**Issue by teocci** [bytedeco/javacv#1102]
I'm writing a multi audio and video stream player in Java and I'm experiencing some latency/delays that mostly and gradually shows up if the application is left running for a while. Sometimes it is after just 2 mins or 20 mins or even after 1 hour later. 

First I thought it was the `SourceDataLine` instantiation and I change it to a mixer from this: `sdl = (SourceDataLine)AudioSystem.getLine(info);` to `sdl = (SourceDataLine)mixer.getLine(info);`. However, it has the same results.

Initially it behaves as expected with a tiny delay of 250 ms approximately. However if the program is left running for a while then that delay is increased to several seconds.

I tested with different headsets. I tested with Java 8, 9, 10 and it consistently displays the same behaviour. I also experimented with drain() and flush() and so on but the only thing that gets rid of the delay is to close and recreate the `SourceDataLine`.

Finally I decided to implement an audio meter  at the moment that I got an audio frame and I notice that what I heard is what audio meter shows. Therefore, the delay is not in the `SourceDataLine` but in the `Frame frame = grabber.grab();` 

Then I check the **ffplay** source code and I found that they have a `sync` option that can be either one of these three options `audio|video|ext`. This option sets the audio-video sync type. 

The `ext` type refers to `AV_SYNC_EXTERNAL_CLOCK` probably this configuration will help to synchronize the stream based on the RTSP clock parameter like in the `ffplay` implementation.

**Comment by saudet** [comment]
JavaCV isn't a media player, so there isn't much room where to include such features. We'd have to basically create that first, which is fine, but someone would need to do it.

**Comment by teocci** [comment]
@saudet yeah I know that. JavaCV is not a player, totally agreed with that. But it is a FFmpeg wrapper that comes with hardware accelerated image display, and optimized to execute code in parallel on multiple cores. That is why I'm implementing an audio player using this awesome tool. However, the problem is that our FFmpegFrameGrabber implementation is based on the time provided by `javacpp.avutil.av_frame_get_best_effort_timestamp`. This method only calculates the elapsed time since the streaming started based on its frame rate. This introduces an increase of the latency if is left running for a while. I think it might be a bug more than an enhancement.

**Comment by saudet** [comment]
It sounds to me like latency increases because of Java Sound though, and we'd have to correct for that as part of your player, for example.

**Comment by teocci** [comment]
I'm using a soft real-time implementation to play the BufferByte i got form the Frame. I test it I don't want you to fix it for me. I also thought that it was a Java Sound issue. After I replace it with an audio mixer I noticed that it is not. That case. I just want to help because JavaCV it was so helpful for me.

**Comment by saudet** [comment]
Ok, I don't understand the issue, but please send a pull request with the fix you are proposing. That's the easiest way to contribute, if you can.
